src/main/java/seedu/address/MainApp.java

```diff
@@ -83,11 +84,17 @@ private Model initModelManager(Storage storage, ReadOnlyUserPrefs userPrefs) {
                 logger.info("Creating a new data file " + storage.getAddressBookFilePath()
                         + " populated with a sample AddressBook.");
             }
-            initialData = addressBookOptional.orElseGet(SampleDataUtil::getSampleAddressBook);
+            // line below can throw SampleDataLoadingException
+            ReadOnlyAddressBook sampleAddressBook = SampleDataUtil.getSampleAddressBook();
+            initialData = addressBookOptional.orElse(sampleAddressBook);
         } catch (DataLoadingException e) {
             logger.warning("Data file at " + storage.getAddressBookFilePath() + " could not be loaded."
                     + " Will be starting with an empty AddressBook.");
             initialData = new AddressBook();
+        } catch (SampleDataLoadingException e) {
```

Comment:
I realised my issue was poorly phrased and might have misled you hahaahha. The sampleData shouldn't have an exception, since we ourselves chose the data within there (unless the user go and modify it, but that one do we really need to defend against)

Reply:
Rather, what I meant was that on a non fresh start (i.e. the addressbook JSON already exists), I can modify the data file to give a VALIDATION_REGEX error.

Reply:
Yeah I understand as well. Just wanted to add the checks in both cases I guess.

Reply:
The addition of the new exception was actually to adding a more specific logger warning to say that something was wrong with the loading of the sample data.